src/solver/conv_hip_implicit_gemm_bwd_data_xdlops.cpp

```diff
@@ -0,0 +1,391 @@
+                                                              args.lPadding,
+                                                              args.rPadding,
+                                                              {},
+                                                              {},
+                                                              {});
+        if(conv_ptrs[i]->IsSupportedArgument(argument_ptr.get()))
+            return true;
+    }
+    return false;
+}
+
+template <typename DataType>
+void ConvHipImplicitGemmBwdXdlops::RunCKSolution(
+    const Handle& handle,
+    const AnyInvokeParams& primitive_parameters,
+    const ProblemDescription& problem,
+    const PerformanceConfigHipImplicitGemmBwdXdlops& config) const
+{
+    const auto args      = CKArgsBwd{problem};
+    const auto conv_ptrs = DeviceOpBwdPtrs<DataType>::GetInstances();
+    auto& conv_ptr       = conv_ptrs.at(config.index);
+    auto& data_ctx       = primitive_parameters.CastTo<conv::DataInvokeParams>();
+    const auto& tensors  = data_ctx.tensors;
+    auto argument_ptr    = conv_ptr->MakeArgumentPointer(
+        static_cast<void*>(tensors.out),
+        const_cast<void*>( // NOLINT (cppcoreguidelines-pro-type-const-cast)
+            static_cast<const void*>(tensors.w)),
+        const_cast<void*>( // NOLINT (cppcoreguidelines-pro-type-const-cast)
+            static_cast<const void*>(tensors.in)),
+        args.N,
+        args.K,
+        args.C,
+        args.input,
+        args.filter,
+        args.output,
+        args.strides,
+        args.dilation,
+        args.lPadding,
+        args.rPadding,
+        {},
+        {},
+        {});
+    auto invoker_ptr            = conv_ptr->MakeInvokerPointer();
+    const auto enable_profiling = handle.IsProfilingEnabled();
+
+    float elapsed_time =
+        invoker_ptr->Run(argument_ptr.get(), {handle.GetStream(), enable_profiling});
+    if(enable_profiling)
+    {
+        handle.ResetKernelTime();
+        handle.AccumKernelTime(elapsed_time);
+    }
+}
+#endif
+
+void PerformanceConfigHipImplicitGemmBwdXdlops::HeuristicInit(const ProblemDescription& problem)
+{
+#if !MIOPEN_BACKEND_HIP || !MIOPEN_USE_COMPOSABLEKERNEL
+    std::ignore = problem;
+#else
+    this->index = 0;
+    switch(problem.conv_problem.GetInDataType())
+    {
+    case miopenHalf: Init<ck::half_t>(problem); break;
+    case miopenFloat: Init<float>(problem); break;
+    case miopenInt8:
+    case miopenInt32:
+    case miopenInt8x4:
+    case miopenBFloat16:
+    case miopenDouble: break;
+    }
+#endif
+}
+
+bool PerformanceConfigHipImplicitGemmBwdXdlops::SetNextValue(const ProblemDescription& problem)
+{
+    if(total_size == -1)
+    {
+        this->HeuristicInit(problem);
+        assert(total_size != -1);
+        return true;
+    }
+    else if((index + 1) < total_size)
+    {
+        ++index;
+        return true;
+    }
+    else
+        return false;
```

Comment:
[Recommendation] We do not "else" in this func I guess
```suggestion
    if((index + 1) < total_size)
    {
        ++index;
        return true;
    }
    return false;
```